pkg/kube/pod/manager.go

```diff
@@ -159,7 +160,10 @@ func (pw *Manager) GetPodByJob(ctx context.Context, job *batch.Job) (*core.Pod,
 	if err != nil {
 		return nil, err
 	}
-	return &podList.Items[0], nil
+	if len(podList.Items) > 0 {
```

Comment:
Any chance podList can be nil?

Reply:
In theory if there's no error it won't be nil. However, we're practitioners so I added nil check to the podList 🤣